.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,153 @@
+  contents: read  # This is required for actions/checkout
+
+jobs:
+  # ensure go.mod and go.sum are updated
+  depscheck:
+    name: Check Dependencies
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Harden Runner
+      uses: step-security/harden-runner@0d381219ddf674d61a7572ddd19d7941e271515c # v2.9.0
+      with:
+        egress-policy: audit
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+    - name: Set up Go
+      uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
+      with:
+        cache: true
+        go-version-file: 'go.mod'
+      id: go
+
+
+    - name: Run 'go mod tidy' and check for differences
+      run: |
+        go mod tidy
+        git diff --exit-code -- go.mod go.sum || \
+        (echo; echo "Unexpected difference in go.mod/go.sum files. Run 'go mod tidy' command or revert any go.mod/go.sum changes and commit."; exit 1)
+
+  # ensure the code builds
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+
+    - name: Harden Runner
+      uses: step-security/harden-runner@0d381219ddf674d61a7572ddd19d7941e271515c # v2.9.0
+      with:
+        egress-policy: audit
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+    - name: Set up Go
+      uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
+      with:
+        cache: true
+        go-version-file: 'go.mod'
+      id: go
+
+    - name: Get dependencies
+      run: |
+        go mod download
+
+    - name: Build
+      run: |
+        go build -v .
+
+#   unit_test:
+#     name: Unit Test
+#     needs: build
+#     runs-on: ubuntu-latest
+#     steps:
+    
+#     - name: Harden Runner
+#       uses: step-security/harden-runner@0d381219ddf674d61a7572ddd19d7941e271515c # v2.9.0
+#       with:
+#         egress-policy: audit
+
+#     - name: Check out code into the Go module directory
+#       uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+#     - name: Set up Go
+#       uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
+#       with:
+#         cache: true
+#         go-version-file: 'go.mod'
+#         id: go
+
+#     # - name: Get dependencies
+#     #   run: |
+#     #     go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.49.0
+#     #     go mod download
+        
+#     - name: Run unit tests
+#       run: |
+#         go test -v ./... -run "^TestUnit" -coverprofile=unit-test-coverage.out
+#         go tool cover -html=unit-test-coverage.out -o unit-test-coverage.cov.html
+
+#     - name: Upload Coverage Artifacts
+#       uses: actions/upload-artifact@0b2256b8c012f0828dc542b3febcab082c67f72b # v4.3.4
+#       with:
+#         name: Test Coverage
+#         path: unit-test-coverage.cov.html
```

Comment:
Do we still need commented out code?

Reply:
this is for running unit test, I suppose we would like to run whole package? delete it or uncomment it?